source/local/local.go

```diff
@@ -144,14 +160,54 @@ func (s *localSource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
-func parseFeatures(lines [][]byte) map[string]string {
+func parseDirectives(line string, opts *parsingOpts) error {
+	lineSplit := strings.SplitN(line, "=", 2)
+
+	// If we have the following directive: # +expiry-time=xxx
+	// we need to get the key '# +expiry-time' and then we need to extract
+	// the following key 'expiry-time'.
+	key := strings.Split(lineSplit[0], DirectivePrefix)[1]
+	if len(lineSplit) == 1 {
+		return ErrorInvalidDirectiveFormat
+	}
+	value := lineSplit[1]
```

Comment:
Using `strings.Split()` for dropping the prefix looks odd and might cause unexpected results in the future (if the value of some directive contained `# +`). How about moving the checking of the prefix from `parseFeatures` over to this `parseDirectives` function?

```suggestion
	if !strings.HasPrefix(line, DirectivePrefix) {
		return nil
	}
	directive := line[len(DirectivePrefix):len(line)]
	split := strings.SplitN(directive, "=", 2)
	key := split[0]
```